internal/controller/nim/account_controller.go

```diff
@@ -383,6 +448,41 @@ func (r *AccountReconciler) reconcileNimPullSecret(
 	return secret, nil
 }
 
+// getSelectedModelList returns a list of Ids of the selected models
+func (r *AccountReconciler) getSelectedModelList(
+	ctx context.Context, cmRef *corev1.ObjectReference,
+	namespace string) ([]string, error) {
+	// if selected model list is not set
+	if cmRef == nil || len(cmRef.Name) == 0 {
+		return []string{}, nil
+	}
+
+	// get the config map that contains the selected model list
+	cmNs := cmRef.Namespace
+	if cmNs == "" {
+		cmNs = namespace
+	}
+	modelListCm := &corev1.ConfigMap{}
+	modelListCmSubject := types.NamespacedName{Name: cmRef.Name, Namespace: cmNs}
+	if err := r.Client.Get(ctx, modelListCmSubject, modelListCm); err != nil {
+		return nil, err
```

Comment:
We should account for NotFound errors here. The statement invoking this function is requeueing for errors, we'll be stuck here forever. We should log, but return an empty slice and a nil to avoid breaking the rest of the flow.

Example scenario: a user modified the Account spec but hadn't yet created the actual ConfigMap or created it with a wrong name.